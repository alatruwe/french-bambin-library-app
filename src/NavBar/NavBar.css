.navbar {
  background-color: antiquewhite;
  display: flex;
  justify-content: flex-end;
}

.navbar-item {
  margin-right: 5%;
}
