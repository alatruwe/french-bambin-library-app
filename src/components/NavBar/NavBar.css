.navbar {
  font-size: calc(5vw + 5px);
}
.nav-links,
.logo {
  text-decoration: none;
  font-family: "Montserrat", sans-serif;
}
.nav-links:visited {
  color: black;
}

.main-nav {
  list-style-type: none;
  text-align: center;
  margin: 15px auto;
  display: none;
}
.logo {
  display: inline-block;
  font-size: 22px;
  margin-top: 10px;
  margin-left: 50px;
  font-size: calc(1rem + 2vw);
}

.navbar-toggle {
  position: absolute;
  top: 10px;
  right: 20px;
  cursor: pointer;
  font-size: 24px;
}

.active {
  display: block;
  text-align: center;
}

ul {
  padding-top: 10px;
}

/* desktop view */
@media screen and (min-width: 1024px) {
  .navbar {
    display: flex;
    justify-content: space-between;
    padding: 20px;
    height: 70px;
    align-items: center;
  }
  .main-nav,
  .active {
    display: flex;
    margin-right: 30px;
    flex-direction: row;
    justify-content: flex-end;
  }

  .nav-links {
    margin: 40px;
    font-size: calc(1rem + 0.7vw);
  }
  .logo {
    margin-top: 0;
  }
  .navbar-toggle {
    display: none;
  }
}
