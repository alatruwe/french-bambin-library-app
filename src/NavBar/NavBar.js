import React, { Component } from "react";
import "./NavBar.css";

class NavBar extends Component {
  render() {
    return (
      <div className="navbar">
        <div className="navbar-item">
          <h2>Log in</h2>
        </div>
        <div className="navbar-item">
          <h2>Sign Up</h2>
        </div>
      </div>
    );
  }
}

export default NavBar;
