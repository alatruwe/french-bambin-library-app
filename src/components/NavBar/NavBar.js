import React, { Component } from "react";
import { Router, NavLink, Link } from "react-router-dom";
import "./NavBar.css";

class NavBar extends Component {
  render() {
    return (
      <nav className="navbar">
        <NavLink exact to="/">
          Home
        </NavLink>
        <Link to="/login">Log in</Link>
        <Link to="/signup">Sign up</Link>
      </nav>
    );
  }
}

export default NavBar;
