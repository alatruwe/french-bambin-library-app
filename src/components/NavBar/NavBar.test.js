import React from "react";
import ReactDOM from "react-dom";
import NavBar from "./NavBar";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`NavBar component`, () => {
  it("renders the component by default", () => {
    const wrapper = shallow(<NavBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
