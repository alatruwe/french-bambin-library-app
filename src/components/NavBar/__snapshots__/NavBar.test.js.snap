// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders the component by default 1`] = `
<nav>
  <div>
    <Link
      to="/signup"
    >
      Sign Up
    </Link>
    <Link
      to="/login"
    >
      Log in
    </Link>
  </div>
</nav>
`;
